CustomQueries/server/aspnet/RevealSdk.Server/Reveal/DataSourceProvider.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Reveal.Sdk;
+using Reveal.Sdk.Data;
+using Reveal.Sdk.Data.Microsoft.SqlServer;
+
+namespace RevealSdk.Server.Reveal
+{
+    public class DataSourceProvider : IRVDataSourceProvider
+    {
+        public Task<RVDataSourceItem> ChangeDataSourceItemAsync(IRVUserContext userContext, string dashboardId, RVDataSourceItem dataSourceItem)
+        {
+            if (dataSourceItem is RVSqlServerDataSourceItem sqlDataSourceItem)
+            {
+                var sqlDataSource = (RVSqlServerDataSource)sqlDataSourceItem.DataSource;
+                UpdateDataSource(sqlDataSource);
+
+                if (sqlDataSourceItem.Id == "MySqlServerDataSourceItem1")
+                {
+                    sqlDataSourceItem.CustomQuery = "SELECT * FROM [Sales].[SalesOrderHeader] WHERE [SalesPersonId] = 279";
```

Comment:
should we show the usage of UserContext here to get the ID and we just hard code the ID on the client?